lib/block-supports/behaviors.php

```diff
@@ -84,17 +84,22 @@ function gutenberg_render_behaviors_support_lightbox( $block_content, $block ) {
 
 	$aria_label = __( 'Enlarge image', 'gutenberg' );
 
+	$processor->next_tag( 'img' );
 	$alt_attribute = $processor->get_attribute( 'alt' );
 
+	// An empty alt attribute `alt=""` is valid for decorative images.
 	if ( null !== $alt_attribute ) {
 		$alt_attribute = trim( $alt_attribute );
 	}
 
+	// It only makes sense to append the alt text to the button aria-label when the alt text is non-empty.
 	if ( $alt_attribute ) {
 		/* translators: %s: Image alt text. */
 		$aria_label = sprintf( __( 'Enlarge image: %s', 'gutenberg' ), $alt_attribute );
 	}
-	$content = $processor->get_updated_html();
+
+	$processor = new WP_HTML_Tag_Processor( $block_content );
```

Comment:
is this line a mistake here? I think we still have the processor above in line 83

Reply:
I'm not familiar with the processor inner workings. To my understanding, after the `next_tag( 'img' )` the processor points to the image and not to the entire chunk of HTML. Would `get_updated_html();` work anyways. Or am I supposed to create a new instance of the processor? I see several instances od the processor being created, I assumed for this reason. Also, the instances are assigner to variables that I'd argue could be better named. Variable names like `$w`, `$m`, `$q` are very confusing and aren't sekf-explanatory. While those short variable names used to be broadely adopted in the early ages of WordPress, I thought that is nwo considered a non-ideal practice. Certainly doesn't help beginners understand how the processor is supposed ot be used.

Reply:
> Would get_updated_html(); work anyways. Or am I supposed to create a new instance of the processor?

Good catch — we no longer need $content here, and we we're not modifying any HTML yet, so we don't need to do the `$processor->get_updated_html()` call either.

> I see several instances od the processor being created, I assumed for this reason.

> Also, the instances are assigner to variables that I'd argue could be better named. Variable names like $w, $m, $q are very confusing and aren't sekf-explanatory.

The other instances of the processor are created because we need to do different processing on the `figure` and `img` elements for each instance. 

Note: We create new instances of the processor for these three scenarios because the processor is [unable to back up or reverse](https://github.com/WordPress/gutenberg/blob/0816b4cdf016f64f580853dbd39c9228eec5c5f4/lib/compat/wordpress-6.3/html-api/class-gutenberg-html-tag-processor-6-3.php#L77).

There is an exception to that though — we could [create a bookmark](https://github.com/WordPress/gutenberg/blob/0816b4cdf016f64f580853dbd39c9228eec5c5f4/lib/compat/wordpress-6.3/html-api/class-gutenberg-html-tag-processor-6-3.php#L171) for the `figure` element and use that to refactor the code for clarity, removing the need for the different instances and cutting down on the variable names if you think that would be clearer. I believe that should work.

Another option: We could rename the `$w`, `$m`, and `$q` variables to `$lightbox_container_processor`, `$initial_image_processor`, and `$enlarged_image_processor`, or something along those lines.

In either case, we can just use `$block_content` for the new instance(s) rather than `$content` — which is [no longer necessary](https://github.com/WordPress/gutenberg/pull/55010/files#r1347947873) — and that should help make the code clearer as well.

Reply:
> I see several instances od the processor being created, I assumed for this reason. Also, the instances are assigner to variables that I'd argue could be better named. Variable names like $w, $m, $q are very confusing and aren't sekf-explanatory. While those short variable names used to be broadely adopted in the early ages of WordPress, I thought that is nwo considered a non-ideal practice. Certainly doesn't help beginners understand how the processor is supposed ot be used.

Yeah this code is ripe for refactoring. Maybe next week @artemiomorales and I can try and do that. The names, the duplicate processors, they can all be improved.

> Would get_updated_html(); work anyways.

@afercia `get_updated_html()` will always return the full HTML document. if nothing has been modified it will be identical to what was passed in when creating the Tag Processor. if modifications have been made, it will return the result of applying all requested changes to that document.

Reply:
@dmsnell Sure, we can do that 👍